code/modules/organs/subtypes/parasite.dm

```diff
@@ -0,0 +1,209 @@
+/obj/item/organ/parasite
+	name = "parasite"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "burst_lie"
+	dead_icon = "bursted_lie"
+
+	organ_tag = "parasite"
+	var/stage = 1
+	var/max_stage = 4
+	var/stage_ticker = 0
+	var/stage_interval = 600 //time between stages, in seconds
+	var/subtle = 0 //will the body reject the parasite naturally?
+
+/obj/item/organ/parasite/process()
+	..()
+
+	if(!owner)
+		return
+
+	if(stage < max_stage)
+		stage_ticker += 2 //process ticks every ~2 seconds
+
+	if(stage_ticker >= stage*stage_interval)
+		stage = min(stage+1,max_stage)
+
+/obj/item/organ/parasite/handle_rejection()
+	if(subtle)
+		return ..()
+	else
+		if(rejecting)
+			rejecting = 0
+		return
+
+///////////////////
+///K'ois Mycosis///
+///////////////////
+
+/obj/item/organ/parasite/kois
+	name = "k'ois mycosis"
+	icon_state = "kois-on"
+	dead_icon = "kois-off"
+
+	organ_tag = "kois"
+
+	parent_organ = "chest"
+
+/obj/item/organ/parasite/kois/process()
```

Comment:
The structure of this proc looks weird. You shouldn't check `stage` for the same value _twice_. As in, you should only have one `if (stage >= 3)` on the same level. _Also_ consider if vs else-if.